tests/integration/suite/sentry/sentry.go

```diff
@@ -0,0 +1,406 @@
+		framework.WithProcesses(m.sentryWithInsecure, m.sentryWithJWKS),
+	}
+}
+
+func (m *insecureValidator) Run(t *testing.T, parentCtx context.Context) {
+	const (
+		defaultAppID     = "myapp"
+		defaultNamespace = "default"
+	)
+
+	// Generate a private privKey that we'll need for tests
+	privKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	require.NoError(t, err, "failed to generate private key")
+
+	// Get a CSR for the app "myapp"
+	defaultAppCSR, err := generateCSR(defaultAppID, privKey)
+	require.NoError(t, err)
+
+	t.Run("insecure validator", func(t *testing.T) {
+		var client sentrypbv1.CAClient
+
+		t.Run("connect to Sentry", func(t *testing.T) {
+			// We need to set up the TLS configuration to validate the TLS certificate provided by Sentry
+			bundle := m.sentryWithInsecure.CABundle()
+			sentrySpiffeID, err := spiffeid.FromString("spiffe://localhost/ns/default/dapr-sentry")
+			require.NoError(t, err, "failed to create Sentry SPIFFE ID")
+			x509bundle, err := x509bundle.Parse(sentrySpiffeID.TrustDomain(), bundle.TrustAnchors)
+			require.NoError(t, err, "failed to create x509 bundle")
+			transportCredentials := grpccredentials.TLSClientCredentials(x509bundle, tlsconfig.AuthorizeID(sentrySpiffeID))
+
+			// Actually establish the connection using gRPC
+			t.Logf("Connecting to Sentry on 127.0.0.1:%d", m.sentryWithInsecure.Port())
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			conn, err := grpc.DialContext(
+				ctx,
+				fmt.Sprintf("127.0.0.1:%d", m.sentryWithInsecure.Port()),
+				grpc.WithTransportCredentials(transportCredentials),
+				grpc.WithReturnConnectionError(),
+				grpc.WithBlock(),
+			)
+			require.NoError(t, err)
+
+			client = sentrypbv1.NewCAClient(conn)
+		})
+
+		t.Run("fails when passing an invalid validator", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			_, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id:             defaultAppID,
+				Namespace:      defaultNamespace,
+				TokenValidator: sentrypbv1.SignCertificateRequest_TokenValidator(-1), // -1 is an invalid enum value
+			})
+			require.Error(t, err)
+
+			grpcStatus, ok := status.FromError(err)
+			require.True(t, ok)
+			require.Equal(t, codes.InvalidArgument, grpcStatus.Code())
+			require.Contains(t, grpcStatus.Message(), "not enabled")
+		})
+
+		t.Run("issue a certificate with insecure validator", func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(parentCtx, 10*time.Second)
+			defer cancel()
+			res, err := client.SignCertificate(ctx, &sentrypbv1.SignCertificateRequest{
+				Id:                        defaultAppID,
+				Namespace:                 defaultNamespace,
+				CertificateSigningRequest: defaultAppCSR,
+				TokenValidator:            sentrypbv1.SignCertificateRequest_INSECURE,
+			})
+			require.NoError(t, err)
+			require.NotEmpty(t, res.WorkloadCertificate)
```

Comment:
Verify the contents of the certificate is correct, i.e. it has been signed by the same Certificate Authority as sentry is serving with, as well as it containing the expected SPIFFE ID of this app.

Reply:
Done